spec/linkedlist.js

```diff
@@ -0,0 +1,238 @@
+      myLinkedList.insertFirst("Mercedes")
+      expect(myLinkedList.find("Ford").data).to.equal("Ford")
+      expect(myLinkedList.find("Porsche")).to.equal(-1)
+    })
+  })
+
+  context('insertBefore( before, after )', () => {
+    it('Inserts a node (with data "before") before the first node containing "after"', () => {
+      const myLinkedList = new LinkedList()
+      myLinkedList.insert("BMW")
+      myLinkedList.insert("Tesla")
+      myLinkedList.insert("Lexus")
+      myLinkedList.insert("Mercedes")
+      myLinkedList.insertBefore("Ford", "BMW")
+      expect(myLinkedList.head.next.data).to.equal("BMW")
+    })
+
+    it('Inserts a node after the first node containing "before"', () => {
+      const myLinkedList = new LinkedList()
+      myLinkedList.insert("BMW")
+      myLinkedList.insert("Tesla")
+      myLinkedList.insert("Lexus")
+      myLinkedList.insert("Mercedes")
+      myLinkedList.insertAfter("BMW", "Bentley")
+      expect(myLinkedList.head.next.data).to.equal("Bentley")
+    })
+  })
+
+  context('insertAfter( before, after )', () => {
+    it('Inserts a node after the first node containing "before"', () => {
+      const myLinkedList = new LinkedList()
+      myLinkedList.insert("BMW")
+      myLinkedList.insert("Tesla")
+      myLinkedList.insert("Lexus")
+      myLinkedList.insert("Mercedes")
+      expect( ()=> myLinkedList.insertAfter("Tesla", "Volkswagen") ).to.alter( ()=> myLinkedList.length(), { from: 4, to: 5 })
+    })
+  })
+
+  context('remove()', () => {
+    it('Remove the last node with a list with many items', () => {
+      const myLinkedList = new LinkedList()
+      myLinkedList.insert("BMW")
+      myLinkedList.insert("Lexus")
+      myLinkedList.insert("Mercedes")
+      myLinkedList.remove()
+      expect(myLinkedList.tail.data).to.equal("Lexus")
+    })
+
+    it('Remove the last node from an empty list', () => {
+      const myLinkedList = new LinkedList()
+      expect(myLinkedList.remove()).to.equal("There is nothing to remove")
+    })
+  })
+
+  context('removeFirst()', () => {
+    it('Remove the first node with a list with many items', () => {
+      const myLinkedList = new LinkedList()
+      myLinkedList.insert("BMW")
+      myLinkedList.insert("Lexus")
+      myLinkedList.insert("Mercedes")
+      myLinkedList.removeFirst()
+      expect(myLinkedList.head.data).to.equal("Lexus")
+    })
+
+    it('Remove the first node from an empty list', () => {
+      const myLinkedList = new LinkedList()
+      expect(myLinkedList.removeFirst()).to.equal("There is nothing to remove")
+    })
+  })
+
+  context('length()', () => {
+    it('Remove the first node with a list with many items', () => {
+      const myLinkedList = new LinkedList()
+      myLinkedList.insert("BMW")
+      myLinkedList.insert("Lexus")
+      myLinkedList.insert("Mercedes")
+      myLinkedList.insert("Tesla")
+      myLinkedList.insert("Ford")
+      expect( myLinkedList.length() ).to.equal(5)
```

Comment:
as long as this is recognizing that this is your method and not the built-in function then you should be fine. I would still add another couple expects to be sure("to alter, is always nice")